core/templates/pages/contributor-dashboard-page/modal-templates/question-suggestion-review-modal.controller.spec.ts

```diff
@@ -335,4 +346,117 @@ describe('Question Suggestion Review Modal Controller', function() {
+        '$uibModalInstance', ['close', 'dismiss']);
+
+      question = QuestionObjectFactory.createFromBackendDict({
+        id: '1',
+        question_state_data: {
+          content: {
+            html: 'Question 1',
+            content_id: 'content_1'
+          },
+          interaction: {
+            answer_groups: [{
+              outcome: {
+                dest: 'outcome 1',
+                feedback: {
+                  content_id: 'content_5',
+                  html: ''
+                },
+                labelled_as_correct: true,
+                param_changes: [],
+                refresher_exploration_id: null
+              },
+              rule_specs: [],
+            }],
+            confirmed_unclassified_answers: [],
+            customization_args: {
+              placeholder: {
+                value: {
+                  content_id: 'ca_placeholder_0',
+                  unicode_str: ''
+                }
+              },
+              rows: { value: 1 }
+            },
+            default_outcome: {
+              dest: null,
+              feedback: {
+                html: 'Correct Answer',
+                content_id: 'content_2'
+              },
+              param_changes: [],
+              labelled_as_correct: true
+            },
+            hints: [{
+              hint_content: {
+                html: 'Hint 1',
+                content_id: 'content_3'
+              }
+            }],
+            solution: {
+              correct_answer: 'This is the correct answer',
+              answer_is_exclusive: false,
+              explanation: {
+                html: 'Solution explanation',
+                content_id: 'content_4'
+              }
+            },
+            id: 'TextInput'
+          },
+          param_changes: [],
+          recorded_voiceovers: {
+            voiceovers_mapping: {}
+          },
+          written_translations: {
+            translations_mapping: {}
+          },
+        },
+      });
+
+      $scope = $rootScope.$new();
+      suggestion = { status: 'rejected' };
+      $controller('QuestionSuggestionReviewModalController', {
+        $scope: $scope,
+        $uibModalInstance: $uibModalInstance,
+        authorName: authorName,
+        contentHtml: contentHtml,
+        misconceptionsBySkill: misconceptionsBySkill,
+        question: question,
+        questionHeader: questionHeader,
+        reviewable: false,
+        skillDifficulty: skillDifficulty,
+        skillRubrics: skillRubrics,
+        suggestion: suggestion,
+        suggestionId: suggestionId
+      });
+    }));
+
+    it('should should fetch the rejection message', function() {
+      let responseDict = {
+        messages: [
+          { text: 'Question submitted.' },
+          { text: 'This is a rejection.' }
+        ]
+      };
+      $httpBackend.expect('GET', '/threadhandler/123').respond(responseDict);
+      $httpBackend.flush();
+      $rootScope.$digest();
```

Comment:
Why is this line needed?

Reply:
This is needed to resolve $q promises (we seem to use this in other spec files where `$httpBackend.flush()` is called) but I tried I just tried to run the test without this and it worked fine. So removing it. Done.